mlir/lib/Gradient/IR/GradientOps.cpp

```diff
@@ -57,12 +57,34 @@ LogicalResult verifyGradInputs(OpState *op_state, func::FuncOp callee, ValueRang
                << "expected " << fnType.getNumInputs() << " but got " << fnArgs.size();
 
     if (callee->getAttrOfType<UnitAttr>(catalyst::quantum::hasInvalidGradientOp)) {
-        return op_state->emitOpError("An operation without a valid gradient was found in code "
-                                     "reachable from the gradient operation.\n"
-                                     "Example of operations not allowed:\n"
-                                     " * mid circuit measurements\n"
-                                     " * callbacks\n"
-                                     " * ZNE mitigation.");
+        // Check that the method is not finite difference, as finite difference should always be
+        // available
+        auto gradOp = dyn_cast<GradOp>(op_state->getOperation());
+        auto jvpOp = dyn_cast<JVPOp>(op_state->getOperation());
+        auto vjpOp = dyn_cast<VJPOp>(op_state->getOperation());
```

Comment:
This is not ideal, but let's keep it for now. I added this ticket https://github.com/PennyLaneAI/catalyst/issues/794 which would allow us to cast all of these into an interface.